pkg/apis/triggers/v1alpha1/trigger_template_validation.go

```diff
@@ -64,3 +69,29 @@ func validateResourceTemplates(templates []TriggerResourceTemplate) *apis.FieldE
 	}
 	return nil
 }
+
+// Verify every param in the ResourceTemplates is declared with a ParamSpec
+func verifyParamDeclarations(params []pipelinev1.ParamSpec, templates []TriggerResourceTemplate) *apis.FieldError {
+	declaredParamNames := map[string]struct{}{}
+	for _, param := range params {
+		declaredParamNames[param.Name] = struct{}{}
+	}
+	paramsRegexp := regexp.MustCompile(`\$\(params.(?P<var>[_a-zA-Z][_a-zA-Z0-9.-]*)\)`)
```

Comment:
Now that I think about it, can we move the declaration out of the function body itself, maybe to a var/const block in the package?

Reply:
Yep, that works for me too. Just made the change @dibyom 👍